### PR TITLE
Fix Bug 1126864: Added Google Tag Manager, Adjusted CSP Policy For Googl...

### PR DESCRIPTION
**Pull request by bensternthal**
...e Tag Manager

Note I adjusted the CSP policy following the same rules as was implemented on affiliates:
https://github.com/mozilla/affiliates/blob/master/affiliates/settings/base.py

I also followed the same template for including the .js file.

### CODE REVIEW

File: masterfirefoxos/settings/base.py [hunk 1]
```diff
@@ -143,12 +147,15 @@
 )
 CSP_SCRIPT_SRC = (
     "'self'",
+    "'unsafe-eval'",
+    "'unsafe-inline'",
```
Comment (jgmize) [inline masterfirefoxos/settings/base.py]: I'm more than a little concerned if these are necessary for GTM to work. These effectively remove all protection from XSS that the CSP is designed to provide.

Comment (bensternthal) [inline masterfirefoxos/settings/base.py]: @Osmose Can you advise here, I copied the permissions from affiliates.

Comment (Osmose) [inline masterfirefoxos/settings/base.py]: IIRC they were required for GTM, and @jgmize is also right in that they render CSP roughly useless. There's a bug open for getting security to take a look at this issue and figure out which way to go: https://bugzilla.mozilla.org/show_bug.cgi?id=1053845

**Comment by glogiotatidis** [comment]
I 'm not clear on how GoogleTagManager is different from GoogleAnalytics, but we did face the exact same issue with the later and CSP is many of our web properties. SUMO, mozillians, flicks and others decided to not inline include Analytics code and instead include it as separate JS file.

This solves the CSP issue and on the downside it might result in inaccurate results in some edge cases. See https://github.com/mozilla/mozillians/pull/500#issuecomment-18964438

**Comment by Osmose** [comment]
IIRC GTM requires these even when included via a separate JS file due to the way it modifies the page, but a double check wouldn't hurt given how long it's been since I last looked at it.

**Comment by jgmize** [comment]
Based on my current understanding of GTM, I don't see how it could work with a CSP that doesn't allow 'unsafe-eval' and 'unsafe-inline'. I would prefer that we load analytics from a separate file without GTM.

**Comment by jgmize** [comment]
After discussing the issues raised above with @bensternthal, @garethc, and @cmore, we decided that the risks of GTM outweighed the benefits for this site, so I'm closing this PR.

**Pull request closed without merging**